src/nested_pandas/nestedframe/core.py

```diff
@@ -274,6 +274,52 @@ def nested_columns(self) -> list:
                 nest_cols.append(column)
         return nest_cols
 
+    def _repr_html_(self) -> str | None:
+        """Override html representation"""
+
+        # Without nested columns, just do representation as normal
+        if len(self.nested_columns) == 0:
+            # This mimics pandas behavior
+            if self.shape[0] > pd.get_option("display.max_rows"):
+                return super().to_html(max_rows=pd.get_option("display.min_rows"), show_dimensions=True)
+            else:
+                return super().to_html(max_rows=pd.get_option("display.max_rows"), show_dimensions=True)
+
+        # Nested Column Formatting
+        # first cell shows the nested df header and a preview row
+        def repack_first_cell(chunk):
+            # Render header separately to keep data aligned
+            output = chunk.head(0).to_html(
+                max_rows=0, max_cols=5, show_dimensions=False, index=False, header=True
+            )
+            # Then add a preview row
+            output += repack_row(chunk)
+            return output
+
+        # remaining cells show only a preview row
+        def repack_row(chunk):
+            return chunk.to_html(max_rows=1, max_cols=5, show_dimensions=True, index=False, header=False)
+
+        # Apply repacking to all nested columns
+        repr = self.style.format(
+            {col: repack_first_cell for col in self.nested_columns}, subset=self.index[0]
+        )
+        repr = repr.format(
+            {col: repack_row for col in self.nested_columns}, subset=pd.IndexSlice[self.index[1] :]
+        )
+
+        # Recover some truncation formatting, limited to head truncation
+        # Use half of pandas sizes to account for large row height
+        if repr.data.shape[0] > pd.get_option("display.max_rows") // 2:
+            html_repr = repr.to_html(max_rows=pd.get_option("display.min_rows") // 2)
+        else:
+            html_repr = repr.to_html(max_rows=0)
```

Comment:
Ok, it's intentional here, I see. So this means "no truncation" when the number of rows is greater than half of `display.max_rows`; correct? (In practice this looks like it also includes row elision, in my sandbox.) Or do you mean to clip it to `1` row or otherwise be minimal?

Reply:
Right, effectively this is just ensuring that all rows are represented if under the `max_rows` threshold, and otherwise a truncated view with `min_rows` is provided. Again removing the `// 2` as per above. I added a comment to explain this more.